src/Symfony/Component/Console/Helper/Tree.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\Helper;
+
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @implements \RecursiveIterator<int, TreeNode>
+ */
+final class Tree implements \RecursiveIterator
+{
+    private readonly TreeStyle $style;
+
+    private \Iterator $childrenIterator;
+
+    public function __construct(
+        private readonly OutputInterface $output,
+        private readonly TreeNode $node,
+        ?TreeStyle $style = null,
+    ) {
+        $this->style = $style ?? TreeStyle::default();
+        $this->childrenIterator = new \IteratorIterator($node->getChildren());
+        $this->childrenIterator->rewind();
+    }
+
+    public function current(): TreeNode
+    {
+        return $this->childrenIterator->current();
+    }
+
+    public function key(): int
+    {
+        return $this->childrenIterator->key();
+    }
+
+    public function next(): void
+    {
+        $this->childrenIterator->next();
+    }
+
+    public function rewind(): void
+    {
+        $this->childrenIterator->rewind();
+    }
+
+    public function valid(): bool
+    {
+        return $this->childrenIterator->valid();
+    }
+
+    public function hasChildren(): bool
+    {
+        if (null === $current = $this->current()) {
+            return false;
+        }
+
+        foreach ($current->getChildren() as $child) {
+            return true;
+        }
+
+        return false;
+    }
+
+    public function getChildren(): \RecursiveIterator
+    {
+        return new self($this->output, $this->current(), $this->style);
+    }
+
+    /**
+     * Recursively renders the tree to the output, applying the tree style.
+     */
+    public function render(): void
+    {
+        $treeIterator = new \RecursiveTreeIterator($this);
+
+        $this->style->applyPrefixes($treeIterator);
+
+        $this->output->writeln($this->node->getValue());
+
+        $visited = new \SplObjectStorage();
+        foreach ($this->traverseWithCycleDetection($treeIterator, $visited) as $line) {
+            $this->output->writeln($line);
+        }
+    }
+
+    /**
+     * Traverses the tree with cycle detection.
+     *
+     * @return \Generator<string>
```

Comment:
I don't think this is right, Jetbrains stubs define the first template argument as TKey and the second one as TYield. You may update it with `\Generator<int, string>` if I get it right (see [the stubs](https://github.com/JetBrains/phpstorm-stubs/blob/03273d078e30d26e5893b49e79bc0bde65585956/standard/_types.php#L156-L163) for ref)

Reply:
Generator even has 4 template types, not just 2, so `\Generator<int, string>` is also wrong (but we could be using `\Traversable<int, string>` if the calling code only care about iteration)